pkg/controllers/management/auth/manager.go

```diff
@@ -218,9 +220,10 @@ func (m *manager) ensureProjectMembershipBinding(roleName, rtbNsAndName, namespa
 
 	if len(objs) == 0 {
 		logrus.Infof("[%v] Creating roleBinding for membership in project %v for subject %v", m.controller, project.Name, subject.Name)
+		rbName := strings.Join([]string{"rolebinding", pkgrbac.HashFromKey(key)}, "-")
 		_, err := m.mgmt.RBAC.RoleBindings(namespace).Create(&v1.RoleBinding{
 			ObjectMeta: metav1.ObjectMeta{
-				GenerateName: "rolebinding-",
+				Name: rbName,
```

Comment:
One question: Since the scope of a roleBinding is limited to a namespace, is it possible to have 2 roleBindings with same Name in different namespaces? If yes then should the hashed key consider namespace as well for roleBindings?

Reply:
Yes, this is possible with the current hashing logic, though I see this as desired functionality. RoleBindings are considered to be duplicates if they have the same: 
1. Subject
2. Role
3. Namespace

But since RoleBindings are namespaced resources, we only need to consider the Subject and Role for the hash. For example: 
```
rb1 = RoleBinding{
  Name: rolebinding-fe3adc2e
  Namespace: ns-1
  Subject: user-1
  Role: member
}
rb2 = RoleBinding{
  Name: rolebinding-fe3adc2e
  Namespace: ns-2
  Subject: user-1
  Role: member
}
```
Here rb1 and rb2 have the same Name due to having the same Subject and Role, but due to being in different namespaces, they will not conflict with each other. 

This is how I've thought through it, but I would not mind getting @dramich input on this as well.

Reply:
NOTE: this also lines up with how deduping is carried out [here](https://github.com/dramich/rancher/blob/2a59dbca0009e2846099e076ee914a56ea1f986f/pkg/agent/clean/binding.go#L231)

Reply:
I've implemented these changes, resolving